src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = !this.hidden;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+
+  bite(herbi) {
+    if (herbi.__proto__ === Herbivore.prototype
```

Comment:
you can use ```instanceof``` method. This operator tests to see if the prototype property of a constructor appears anywhere in the prototype chain of an object.

Reply:
agree, use instanceof